In [6]:
import kagglehub
from kagglehub import KaggleDatasetAdapter
import pandas as pd

In [21]:
file_path = "./alzheimers_prediction_dataset.csv"
df = pd.read_csv(file_path)
df.head(5)

Country  Age  Gender  Education Level   BMI Physical Activity Level  \
0         Spain   90    Male                1  33.0                  Medium   
1     Argentina   72    Male                7  29.9                  Medium   
2  South Africa   86  Female               19  22.9                    High   
3         China   53    Male               17  31.2                     Low   
4        Sweden   58  Female                3  30.0                    High   

  Smoking Status Alcohol Consumption Diabetes Hypertension  ...  \
0          Never        Occasionally       No           No  ...   
1         Former               Never       No           No  ...   
2        Current        Occasionally       No          Yes  ...   
3          Never           Regularly      Yes           No  ...   
4         Former               Never      Yes           No  ...   

  Dietary Habits Air Pollution Exposure  Employment Status Marital Status  \
0        Healthy                   High            Retired         Single   
1        Healthy                 Medium         Unemployed        Widowed   
2        Average                 Medium           Employed         Single   
3        Healthy                 Medium            Retired         Single   
4      Unhealthy                   High           Employed        Married   

  Genetic Risk Factor (APOE-ε4 allele) Social Engagement Level Income Level  \
0                                   No                     Low       Medium   
1                                   No                    High          Low   
2                                   No                     Low       Medium   
3                                   No                    High       Medium   
4                                   No                     Low       Medium   

  Stress Levels Urban vs Rural Living Alzheimer’s Diagnosis  
0          High                 Urban                    No  
1          High                 Urban                    No  
2          High                 Rural                    No  
3           Low                 Rural                    No  
4          High                 Rural                    No  

[5 rows x 25 columns]

In [10]:
df.shape

(74283, 25)

In [11]:
df.isnull().sum()

Country                                 0
Age                                     0
Gender                                  0
Education Level                         0
BMI                                     0
Physical Activity Level                 0
Smoking Status                          0
Alcohol Consumption                     0
Diabetes                                0
Hypertension                            0
Cholesterol Level                       0
Family History of Alzheimer’s           0
Cognitive Test Score                    0
Depression Level                        0
Sleep Quality                           0
Dietary Habits                          0
Air Pollution Exposure                  0
Employment Status                       0
Marital Status                          0
Genetic Risk Factor (APOE-ε4 allele)    0
Social Engagement Level                 0
Income Level                            0
Stress Levels                           0
Urban vs Rural Living             

In [13]:
duplicated = df[df.duplicated()]
duplicated

Empty DataFrame
Columns: [Country, Age, Gender, Education Level, BMI, Physical Activity Level, Smoking Status, Alcohol Consumption, Diabetes, Hypertension, Cholesterol Level, Family History of Alzheimer’s, Cognitive Test Score, Depression Level, Sleep Quality, Dietary Habits, Air Pollution Exposure, Employment Status, Marital Status, Genetic Risk Factor (APOE-ε4 allele), Social Engagement Level, Income Level, Stress Levels, Urban vs Rural Living, Alzheimer’s Diagnosis]
Index: []

[0 rows x 25 columns]

In [20]:
df.groupby(df.columns[-1]).count()

Country    Age  Gender  Education Level    BMI  \
Alzheimer’s Diagnosis                                                   
No                       43570  43570   43570            43570  43570   
Yes                      30713  30713   30713            30713  30713   

                       Physical Activity Level  Smoking Status  \
Alzheimer’s Diagnosis                                            
No                                       43570           43570   
Yes                                      30713           30713   

                       Alcohol Consumption  Diabetes  Hypertension  ...  \
Alzheimer’s Diagnosis                                               ...   
No                                   43570     43570         43570  ...   
Yes                                  30713     30713         30713  ...   

                       Sleep Quality  Dietary Habits  Air Pollution Exposure  \
Alzheimer’s Diagnosis                                                          
No                             43570           43570                   43570   
Yes                            30713           30713                   30713   

                       Employment Status  Marital Status  \
Alzheimer’s Diagnosis                                      
No                                 43570           43570   
Yes                                30713           30713   

                       Genetic Risk Factor (APOE-ε4 allele)  \
Alzheimer’s Diagnosis                                         
No                                                    43570   
Yes                                                   30713   

                       Social Engagement Level  Income Level  Stress Levels  \
Alzheimer’s Diagnosis                                                         
No                                       43570         43570          43570   
Yes                                      30713         30713          30713   

                       Urban vs Rural Living  
Alzheimer’s Diagnosis                         
No                                     43570  
Yes                                    30713  

[2 rows x 24 columns]

There are 2 types of possible results:
* Yes with 30713 rows
* No with 43570 rows


In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74283 entries, 0 to 74282
Data columns (total 25 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   Country                               74283 non-null  object 
 1   Age                                   74283 non-null  int64  
 2   Gender                                74283 non-null  object 
 3   Education Level                       74283 non-null  int64  
 4   BMI                                   74283 non-null  float64
 5   Physical Activity Level               74283 non-null  object 
 6   Smoking Status                        74283 non-null  object 
 7   Alcohol Consumption                   74283 non-null  object 
 8   Diabetes                              74283 non-null  object 
 9   Hypertension                          74283 non-null  object 
 10  Cholesterol Level                     74283 non-null  object 
 11  Family History 